src/NuGet.Core/NuGet.Packaging/PackageExtraction/StreamExtensions.cs

```diff
@@ -5,12 +5,20 @@
 using System.Buffers;
 using System.IO;
 using System.IO.MemoryMappedFiles;
+using System.Threading.Tasks;
+using System.Threading;
 using NuGet.Common;
 
 namespace NuGet.Packaging
 {
     public static class StreamExtensions
     {
+        public static readonly int BufferSize = 8192;
+        public static async Task CopyToAsync(this Stream stream, Stream destination, CancellationToken token)
```

Comment:
Can we compile this conditionally only for the frameworks that need it?
For example, .NET 5 has it, no need for us to declare extensions for that.

Reply:
`Stream.CopyToAsync()` has an overloaded method that accepts `CancellationToken` both in .NET 472 and .NET 5. The only difference is .NET 5 has [CopyToAsync(Stream, CancellationToken)](https://learn.microsoft.com/en-us/dotnet/api/system.io.stream.copytoasync?view=net-5.0#system-io-stream-copytoasync(system-io-stream-system-threading-cancellationtoken)) but .NET 472 does not have this overload. Similar approach followed in `NuGet.Protocol.StreamExtensions`.
https://github.com/NuGet/NuGet.Client/blob/dcb355d1a989be369e5ac3e3cd6eeea449a4b5eb/src/NuGet.Core/NuGet.Protocol/Utility/StreamExtensions.cs#L16-L19

Reply:
What's your proposal in this case @kartheekp-ms 

I'm not sure what you said makes my suggestion invalid.

Reply:
Your suggestion is valid. We can add a conditional compilation so that the extension method is called just for `.NET Framework 472`. I tried to explain the differences in the API between .NET 5 and .NET Framework and the current approach that we follow in `NuGet.Protocol` library.

Reply:
I see. 

My idea is that by not making it part of the public API, if at some point we migrate away from netstandard2.0 and net472, we can simply not have the redudant extension method anymore.

Reply:
If it needs to be shared in more than 1 project, I like https://github.com/NuGet/NuGet.Client/tree/dev/build/Shared for those purposes :D
Especially if we add conditional compilation.